src/Mod/PartDesign/Gui/TaskHoleParameters.ui

```diff
@@ -128,30 +128,30 @@
      </property>
      <property name="maximumSize">
       <size>
-       <width>140</width>
+       <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
     </widget>
    </item>
-   <item row="6" column="3">
+   <item row="6" column="2">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>Clearance</string>
      </property>
     </widget>
    </item>
-   <item row="6" column="5">
+   <item row="6" column="4">
     <widget class="QComboBox" name="ThreadFit">
```

Comment:
> See bug B reported here:

This still doesn't seem to work. When activating the Custom values option the default Hole cut diameter is set to 1.3 mm. When I change it the view updates but after pressing OK it jumps back to 1.3 mm. When changing this value directly in the property editor it accepts the value. But when double-clicking the hole feature the value jumps back to 1.3 mm.

Reply:
> There were 2 issues: the diameter widget had a fixed maximum and there was a superfluous spacer leading to unnecessary whitespace

It looks much better now but I don't know if it works on all systems correctly now. So, changing the size policy of the labels, the combo box and spin box should fix it:
set the horizontal size policy of the labels to Maximum and that of the other widgets to Expanding

Then the labels don't change their widths while the other two widgets take all available space.

Reply:
> 6) Observe that the hole is not through all, but only 10 meter deep. 

You don't claim to fix this but you once said:

> Concerning E, this is a restriction of OCC, however, this is not yet documented in the Wiki -> I'll do this later today.

I can't see why this should be an OCC limitation. Inside the source code there is this block:
```
        else if ( method == "ThroughAll" ) {
            // Use a large (10m), but finite length
            length = 1e4;
        }
```
I can imagine that the idea of the original author (04b176ebfb) was to have a valid number and thought that we never work in the dimensions of 10000. It can be easily fixed with
```
            Bnd_Box box;
            BRepBndLib::Add(base, box);
            box.SetGap(0.0);
            length = sqrt(box.SquareExtent());
```

Reply:
>             Bnd_Box box;
>             BRepBndLib::Add(base, box);
>             box.SetGap(0.0);
>             length = sqrt(box.SquareExtent());

Exactly! But note that you should also add the sketch profile to the BB. In case it is not attached directly to the body.

Reply:
> It can be easily fixed with
> ```
>             Bnd_Box box;
>             BRepBndLib::Add(base, box);
>             box.SetGap(0.0);
>             length = sqrt(box.SquareExtent());
> ```

Good. However, let's put this aside since I have problems to keep on track. So lets fix the different issues step by step. This one here should go into a separate PR.

Reply:
> set the horizontal size policy of the labels to Maximum and that of the other widgets to Expanding

Thanks, I did this in PR #4343 .

Reply:
> This still doesn't seem to work. When activating the Custom values option the default Hole cut diameter is set to 1.3 mm. When I change it the view updates but after pressing OK it jumps back to 1.3 mm. When changing this value directly in the property editor it accepts the value. But when double-clicking the hole feature the value jumps back to 1.3 mm.

This and the other cut issues are fixed with PR #4347